File: Factory/FactoryMethod/Java/src/Pizza/NewYorkStyle/NewYorkStylePepperoniPizza.java
package eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class NewYorkStylePepperoniPizza extends Pizza {
    public NewYorkStylePepperoniPizza(){
        name = "New York Style Pepperoni Pizza";
        dough = "Think Crust Dough";
        sauce = "Marinara Sauce";

        toppings.add("Grated Reggiano Cheese");
        toppings.add("Sliced Pepperoni");
        toppings.add("Garlic");
        toppings.add("Onion");
        toppings.add("Mushrooms");
        toppings.add("Red Pepper");
    }
}

File: Factory/FactoryMethod/TypeScript/src/Pizza/NewYorkStyle/NewYorkStylePepperoniPizza.ts
import Pizza from '../Pizza';

class NewYorkStylePepperoniPizza extends Pizza {
  constructor() {
    super();
    this.name = 'New York Style Pepperoni Pizza';
    this.dough = 'Think Crust Dough';
    this.sauce = 'Marinara Sauce';

    this.toppings.push('Grated Reggiano Cheese');
    this.toppings.push('Sliced Pepperoni');
    this.toppings.push('Garlic');
    this.toppings.push('Onion');
    this.toppings.push('Mushrooms');
    this.toppings.push('Red Pepper');
  }
}

export default NewYorkStylePepperoniPizza;
